### DISCUSSION TITLE
Pinocchio meets do-mpc

### DISCUSSION THREAD

**Discussion by AleksandrKashirin** [stack-of-tasks/pinocchio#2002] [Q&A]
Hi, dear community!
I am novice to this library!
However I am trying to connect Pinocchio to the [do-mpc](https://www.do-mpc.com/en/latest/) library.
Can anyone please explain if its possible to extract the equations of nonlinear system dynamics in analytical form for example of cart-pendulum system, so I can use these equations in do-mpc library to solve optimization problem? 

Also, I see that Pinocchio is friendly to CasADi so as do-mpc. Maybe there is an already good way to connect them that I cannot find.

Greatly appreciate any advice,
Thank you!

-Aleksandr

**Comment by jcarpent** [comment]
It will soon be possible to use Casasi or CppAD in Python to fully extract the computational graph of Pinocchio and plug it into optimization frameworks like Acado, do-mpc, croccodyl, etc.
